sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/concatenate_data/cloud_collector.hpp

```diff
@@ -0,0 +1,116 @@
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * $Id: concatenate_data.cpp 35231 2011-01-14 05:33:20Z rusu $
+ *
+ */
+
+#ifndef AUTOWARE__POINTCLOUD_PREPROCESSOR__CONCATENATE_DATA__CLOUD_COLLECTOR_HPP_
+#define AUTOWARE__POINTCLOUD_PREPROCESSOR__CONCATENATE_DATA__CLOUD_COLLECTOR_HPP_
+
+#include "combine_cloud_handler.hpp"
+
+#include <list>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+namespace autoware::pointcloud_preprocessor
+{
+
+class PointCloudConcatenateDataSynchronizerComponent;
+class CombineCloudHandler;
+
+class CloudCollector
+{
+public:
+  CloudCollector(
+    std::shared_ptr<PointCloudConcatenateDataSynchronizerComponent> concatenate_node,
+    std::list<std::shared_ptr<CloudCollector>> & collectors,
+    std::shared_ptr<CombineCloudHandler> combine_cloud_handler, int num_of_clouds, double time);
+
+  void setReferenceTimeStamp(double timestamp, double noise_window);
+  std::tuple<double, double> getReferenceTimeStampBoundary();
+  void processCloud(std::string topic_name, sensor_msgs::msg::PointCloud2::SharedPtr cloud);
+  void concatenateCallback();
+  std::tuple<
+    sensor_msgs::msg::PointCloud2::SharedPtr,
+    std::unordered_map<std::string, sensor_msgs::msg::PointCloud2::SharedPtr>,
+    std::unordered_map<std::string, double>>
+  concatenateClouds(
+    std::unordered_map<std::string, sensor_msgs::msg::PointCloud2::SharedPtr> topic_to_cloud_map);
+
+  void publishClouds(
+    sensor_msgs::msg::PointCloud2::SharedPtr concatenate_cloud_ptr,
+    std::unordered_map<std::string, sensor_msgs::msg::PointCloud2::SharedPtr>
+      topic_to_transformed_cloud_map,
+    std::unordered_map<std::string, double> topic_to_original_stamp_map);
+
+  void deleteCollector();
+
+  std::unordered_map<std::string, sensor_msgs::msg::PointCloud2::SharedPtr> getTopicToCloudMap();
+
+private:
+  std::shared_ptr<PointCloudConcatenateDataSynchronizerComponent> concatenate_node_;
+  std::list<std::shared_ptr<CloudCollector>> & collectors_;
+  std::shared_ptr<CombineCloudHandler> combine_cloud_handler_;
+  rclcpp::TimerBase::SharedPtr timer_;
+  std::unordered_map<std::string, sensor_msgs::msg::PointCloud2::SharedPtr> topic_to_cloud_map_;
+  uint64_t num_of_clouds_;
+  double timeout_sec_;
+  double reference_timestamp_min_;
+  double reference_timestamp_max_;
+  std::mutex mutex_;
```

Comment:
Please choose a more descriptive name that includes the purpose of the mutex, or the name of the member variable this mutex protects.

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/8300/commits/eeb310dcef5c7bfc8332e8f25af23e9bc5a5c003